mora/service/address.py

```diff
@@ -51,24 +51,26 @@
                             url_prefix='/service')
 
 
-def get_relation_for(classobj, value):
-    scope = classobj['scope']
-
-    if not isinstance(value, str):
-        value = str(value)
+def get_relation_for(classobj, addrobj, fallback=None):
+    scope = common.checked_get(classobj, 'scope', '', required=True)
 
     if scope == 'DAR':
-        if not util.is_uuid(value):
-            raise ValueError(
-                '{!r} is not a valid address UUID!'.format(value),
-            )
-
-        r = {
-            'uuid': value,
+        return {
+            'uuid': common.get_uuid(addrobj, fallback),
             'objekttype': classobj['uuid'],
         }
 
     elif scope in URN_PREFIXES:
+        if (
+            'urn' in addrobj or
+            keys.VALUE not in addrobj and fallback and 'urn' in fallback
+        ):
+            return {
+                'urn': common.get_urn(addrobj, fallback),
+                'objekttype': classobj['uuid'],
+            }
+
+        value = common.checked_get(addrobj, keys.VALUE, '', required=True)
```

Comment:
Shouldn't we use fallback here as well?

Reply:
> Shouldn't we use fallback here as well?

The line above constitutes the fallback — I'll add a comment to that effect.

Reply:
Right, having read the comment and the function one more time, I think it makes sense. This function could really use a refactoring or some more comments. Is the structure and the logic of the address object documented anywhere? Maybe we should refer to that from here...